### PR TITLE
Resolve Tabs style checks

### PR DESCRIPTION
**Pull request by Imericxu**
#### What is it?

- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Clean up equals methods
- Add hashCode methods matching respective equals to resolve style check warnings

#### APK testing 

On the website, the APK can be found by going to the "Checks" tab below the title and then on "artifacts" on the right.

#### Due diligence

- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Imericxu** [comment]
I see that my pull request has a lot more commits than other people’s; should I squash them?

**Comment by Stypox** [comment]
@Imericxu no, that's ok

File: app/src/main/java/org/schabi/newpipe/settings/tabs/Tab.java [hunk 1]
```diff
@@ -576,15 +600,34 @@ protected void readDataFromJson(final JsonObject jsonObject) {
...
+
+            if (!(super.equals(obj))) {
+                // Base objects are different
+                return false;
             }
 
-            return (playlistId == ((PlaylistTab) obj).playlistId)                     // local
-                    || (playlistServiceId == ((PlaylistTab) obj).playlistServiceId    // remote
-                    && Objects.equals(playlistUrl, ((PlaylistTab) obj).playlistUrl));
+            return playlistServiceId == other.playlistServiceId // Remote
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/settings/tabs/Tab.java:614]: Let's keep it consistent
```suggestion
            return super.equals(obj)
                    && playlistServiceId == other.playlistServiceId // Remote
```

File: app/src/main/java/org/schabi/newpipe/settings/tabs/Tab.java [hunk 2]
```diff
@@ -576,15 +600,34 @@ protected void readDataFromJson(final JsonObject jsonObject) {
...
+            if (!(obj instanceof PlaylistTab)) {
+                return false;
+            }
+
+            final PlaylistTab other = (PlaylistTab) obj;
+
+            if (!(super.equals(obj))) {
+                // Base objects are different
+                return false;
             }
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/settings/tabs/Tab.java:612]:
```suggestion
```

**Review by Stypox — CHANGES_REQUESTED** [review]
Looks mostly good, thank you!

**Review by Stypox — APPROVED** [review]
There are now only three checkstyle warnings. Great :-)

**Pull request merged**